modules/javafx.controls/src/main/java/javafx/scene/control/skin/VirtualFlow.java

```diff
@@ -1458,12 +1458,44 @@ public void scrollTo(int index) {
         if (cell != null) {
             scrollTo(cell);
         } else {
+            //see JDK-8197536
+            if(tryScrollOneUpOrDown(index, true))
+                return;
+            else if (tryScrollOneUpOrDown(index, false))
+                return;
+
             adjustPositionToIndex(index);
             addAllToPile();
             requestLayout();
         }
     }
 
+    //will return true if scroll is successful
+    private boolean tryScrollOneUpOrDown(int targetIndex, boolean down) {
+        //if going down, cell diff is -1, because it will get the target cell index and check if previous
+        //cell is visible to base the position
+        int indexDiff = (down) ? -1 : 1;
```

Comment:
Nit-pick: I would remove the unnecessary parentheses around `(down)`.

Reply:
done!